cpp/dolfinx/fem/assemble_matrix_impl.h

```diff
@@ -173,11 +173,11 @@ void assemble_exterior_facets(
   const int ndim1 = bs1 * num_dofs1;
   std::vector<T> Ae(ndim0 * ndim1);
   const xtl::span<T> _Ae(Ae);
-
-  for (std::size_t index = 0; index < facets.size(); ++index)
+  assert(facets.size() % 2 == 0);
+  for (std::size_t index = 0; index < facets.size() / 2; ++index)
```

Comment:
`for (std::size_t index = 0; index < facets.size(); index += 2)` and simplifications below would be neater.